src/analysis/seminorm.lean

```diff
@@ -278,6 +278,61 @@ variables (p : seminorm 𝕜 E) (c : 𝕜) (x y : E) (r : ℝ)
 protected lemma smul : p (c • x) = ∥c∥ * p x := p.smul' _ _
 protected lemma triangle : p (x + y) ≤ p x + p y := p.triangle' _ _
 
+instance : has_add (seminorm 𝕜 E) :=
+{ add := λ p q,
+  { to_fun := λ x, p x + q x,
+    smul' := λ a x, by rw [p.smul, q.smul, mul_add],
+    triangle' := λ _ _, has_le.le.trans_eq (add_le_add (p.triangle _ _) (q.triangle _ _))
+      (add_add_add_comm _ _ _ _) } }
+
+lemma coe_add (p q : seminorm 𝕜 E) : ⇑(p + q) = p + q := rfl
+
+@[simp] lemma add_apply (p q : seminorm 𝕜 E) (x : E) : (p + q) x = p x + q x := rfl
+
+instance : ordered_cancel_add_comm_monoid (seminorm 𝕜 E) :=
+fun_like.coe_injective.ordered_cancel_add_comm_monoid _ rfl coe_add
```

Comment:
Can you add the `add_monoid` instance we discussed in the zulip thread using `add_monoid_smul` too? Otherwise there's a non-defeq diamond in the nat action

Reply:
I forgot that there might be a diamond. Added it.